payment/controllers.go

```diff
@@ -500,28 +503,101 @@ func GetOrderCreds(service *Service) handlers.AppHandler {
 			)
 		}
 
-		creds, err := service.Datastore.GetOrderCreds(*orderID.UUID(), false)
+		var credentialType string
+		const (
+			singleUse   = "single-use"
+			timeLimited = "time-limited"
+		)
+
+		// get the order from datastore
+		order, err := service.Datastore.GetOrder(*orderID.UUID())
 		if err != nil {
-			return handlers.WrapError(err, "Error getting claim", http.StatusBadRequest)
+			return handlers.WrapError(
+				err,
+				"failed to get order", http.StatusInternalServerError)
+		}
+
+		// look through order, find out what all the order item's credential types are
+		for i, v := range order.Items {
+			if i > 0 {
+				if v.CredentialType != credentialType {
+					// all the order items on the order need the same credential type
+					return handlers.WrapError(
+						fmt.Errorf("all order item credential types must converge"),
+						"failed to get credentials", http.StatusConflict)
+				}
+			} else {
+				credentialType = v.CredentialType
+			}
 		}
 
-		if creds == nil {
-			return &handlers.AppError{
-				Message: "Credentials do not exist",
-				Code:    http.StatusNotFound,
-				Data:    map[string]interface{}{},
+		if credentialType == singleUse {
+			creds, err := service.Datastore.GetOrderCreds(*orderID.UUID(), false)
+			if err != nil {
+				return handlers.WrapError(err, "Error getting claim", http.StatusBadRequest)
 			}
-		}
 
-		status := http.StatusOK
-		for i := 0; i < len(*creds); i++ {
-			if (*creds)[i].SignedCreds == nil {
-				status = http.StatusAccepted
-				break
+			if creds == nil {
+				return &handlers.AppError{
+					Message: "Credentials do not exist",
+					Code:    http.StatusNotFound,
+					Data:    map[string]interface{}{},
+				}
 			}
-		}
 
-		return handlers.RenderContent(r.Context(), creds, w, status)
+			status := http.StatusOK
+			for i := 0; i < len(*creds); i++ {
+				if (*creds)[i].SignedCreds == nil {
+					status = http.StatusAccepted
+					break
+				}
+			}
+
+			return handlers.RenderContent(r.Context(), creds, w, status)
+		} else if credentialType == timeLimited {
+			// only brave merchant
+			if !order.Location.Valid || order.Location.String != "brave.com" {
+				return handlers.WrapError(
+					fmt.Errorf("failed to create time limited credentials"),
+					"only the Brave merchant can have time limited credentials", http.StatusBadRequest)
+			}
+			// is the order paid?
+			if !order.IsPaid() {
+				return handlers.WrapError(
+					fmt.Errorf("order is not paid"),
+					"cannot get credentials for unpaid orders", http.StatusBadRequest)
+			}
+
+			issuedAt := order.CreatedAt
+			expiresAt := issuedAt.AddDate(0, 0, 35)
+
+			var credentials []TimeLimitedCreds
+			timeLimitedSecret := cryptography.NewTimeLimitedSecret([]byte(os.Getenv("BRAVE_MERCHANT_KEY")))
+
+			for _, item := range order.Items {
+				// iterate through order items, derive the time limited creds
+				timeBasedToken, err := timeLimitedSecret.Derive(issuedAt, expiresAt)
```

Comment:
this doesn't appear to depend on the item, we should ensure time limited credentials for one item cannot be redeemed against another

Reply:
fixed here https://github.com/brave-intl/bat-go/pull/863/commits/efe0e5b438741bbde13d803c3ba69ec54682aa72